protocols/autonat/src/protocol.rs

```diff
@@ -227,10 +229,12 @@ impl DialResponse {
 
         Ok(match msg.dial_response {
             Some(structs_proto::message::DialResponse {
-                status: Some(0),
+                status: Some(status),
                 status_text,
                 addr: Some(addr),
-            }) => {
+            }) if structs_proto::message::ResponseStatus::from_i32(status)
```

Comment:
Where does `from_i32` come from? It's not from `TryFrom<i32>`, right?

Reply:
Prost generates the following code:

``` rust
    #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, PartialOrd, Ord, ::prost::Enumeration)]
    #[repr(i32)]
    pub enum ResponseStatus {
        Ok = 0,
        EDialError = 100,
        EDialRefused = 101,
        EBadRequest = 200,
        EInternalError = 300,
    }
```

The `Enumeration` derive macro then generates the `from_i32` implementation:

https://docs.rs/prost-derive/latest/src/prost_derive/lib.rs.html#299-304

Reply:
Ahh, I didn't know that yet about prost. Thanks for the explanation!